### ISSUE TITLE
[JENKINS-34174] Jenkins 2.0-rc1 - Getting started hangs installing plugins

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#17018]
<p>Jenkins 2.0-rc1 hangs for me when installing plugins. I started it with a new jenkins home:</p>

<p>set JENKINS_HOME=c:\j2<br/>
java -jar jenkins-2.0-rc1.war<br/>
Copy admin password into localhost:8080<br/>
Install suggested plugins</p>

<p>Some plugins install, others just sit and spin. I have left if for over an hour.</p>

<p>There is an exception in the console:</p>

<pre>
java.lang.InstantiationException: java.lang.NoClassDefFoundError: com/cloudbees/hudson/plugins/folder/AbstractFolderPropertyDescriptor
at net.java.sezpoz.IndexItem.element(IndexItem.java:146)
at hudson.ExtensionFinder$Sezpoz._find(ExtensionFinder.java:623)
at hudson.ExtensionFinder$Sezpoz.access$800(ExtensionFinder.java:550)
at hudson.ExtensionFinder$Sezpoz$1.find(ExtensionFinder.java:594)
at hudson.ExtensionFinder$GuiceFinder.refresh(ExtensionFinder.java:329)
at jenkins.model.Jenkins.refreshExtensions(Jenkins.java:2383)
at hudson.PluginManager.dynamicLoad(PluginManager.java:691)
at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:1670)
at hudson.model.UpdateCenter$DownloadJob.run(UpdateCenter.java:1452)
at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
at java.util.concurrent.FutureTask.run(Unknown Source)
at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:110)
at java.lang.<span class="code-object">Thread</span>.run(Unknown Source)
Caused by: java.lang.NoClassDefFoundError: com/cloudbees/hudson/plugins/folder/A bstractFolderPropertyDescriptor
at java.lang.<span class="code-object">ClassLoader</span>.defineClass1(Native Method)
at java.lang.<span class="code-object">ClassLoader</span>.defineClass(Unknown Source)
at jenkins.util.AntClassLoader.defineClassFromData(AntClassLoader.java:1139)
at hudson.ClassicPluginStrategy$AntClassLoader2.defineClassFromData(ClassicPluginStrategy.java:895)
at jenkins.util.AntClassLoader.getClassFromStream(AntClassLoader.java:1310)
at jenkins.util.AntClassLoader.findClassInComponents(AntClassLoader.java:1366)
at jenkins.util.AntClassLoader.findClass(AntClassLoader.java:1326)
at sun.reflect.GeneratedMethodAccessor12.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at jenkins.ClassLoaderReflectionToolkit.invoke(ClassLoaderReflectionToolkit.java:44)
at jenkins.ClassLoaderReflectionToolkit._findClass(ClassLoaderReflectionToolkit.java:86)
at hudson.PluginManager$UberClassLoader.findClass(PluginManager.java:1548)
at java.lang.<span class="code-object">ClassLoader</span>.loadClass(Unknown Source)
at java.lang.<span class="code-object">ClassLoader</span>.loadClass(Unknown Source)
at net.java.sezpoz.IndexItem.element(IndexItem.java:134)
... 12 more
Caused by: java.lang.ClassNotFoundException: com.cloudbees.hudson.plugins.folder.AbstractFolderPropertyDescriptor
at jenkins.util.AntClassLoader.findClassInComponents(AntClassLoader.java:1376)
at jenkins.util.AntClassLoader.findClass(AntClassLoader.java:1326)
at jenkins.util.AntClassLoader.loadClass(AntClassLoader.java:1079)
at java.lang.<span class="code-object">ClassLoader</span>.loadClass(Unknown Source)
... 28 more
</pre>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rg">rg</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174" target="_blank">Jenkins 2.0-rc1 - Getting started hangs installing plugins</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kzantow.png" title="kzantow's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kzantow">kzantow</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>label(s)</b>: testfest
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-04-26T14:50:50+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Jenkins 2.0-rc1 hangs for me when installing plugins. I started it with a new jenkins home:</p>

<p>set JENKINS_HOME=c:\j2
java -jar jenkins-2.0-rc1.war
Copy admin password into localhost:8080
Install suggested plugins</p>

<p>Some plugins install, others just sit and spin. I have left if for over an hour.</p>

<p>There is an exception in the console:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">java.lang.InstantiationException: java.lang.NoClassDefFoundError: com/cloudbees/hudson/plugins/folder/AbstractFolderPropertyDescriptor
        at net.java.sezpoz.IndexItem.element(IndexItem.java:146)
        at hudson.ExtensionFinder$Sezpoz._find(ExtensionFinder.java:623)
        at hudson.ExtensionFinder$Sezpoz.access$800(ExtensionFinder.java:550)
        at hudson.ExtensionFinder$Sezpoz$1.find(ExtensionFinder.java:594)
        at hudson.ExtensionFinder$GuiceFinder.refresh(ExtensionFinder.java:329)
        at jenkins.model.Jenkins.refreshExtensions(Jenkins.java:2383)
        at hudson.PluginManager.dynamicLoad(PluginManager.java:691)
        at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:1670)
        at hudson.model.UpdateCenter$DownloadJob.run(UpdateCenter.java:1452)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:110)
        at java.lang.<span class="code-object">Thread</span>.run(Unknown Source)
Caused by: java.lang.NoClassDefFoundError: com/cloudbees/hudson/plugins/folder/A bstractFolderPropertyDescriptor
        at java.lang.<span class="code-object">ClassLoader</span>.defineClass1(Native Method)
        at java.lang.<span class="code-object">ClassLoader</span>.defineClass(Unknown Source)
        at jenkins.util.AntClassLoader.defineClassFromData(AntClassLoader.java:1139)
        at hudson.ClassicPluginStrategy$AntClassLoader2.defineClassFromData(ClassicPluginStrategy.java:895)
        at jenkins.util.AntClassLoader.getClassFromStream(AntClassLoader.java:1310)
        at jenkins.util.AntClassLoader.findClassInComponents(AntClassLoader.java:1366)
        at jenkins.util.AntClassLoader.findClass(AntClassLoader.java:1326)
        at sun.reflect.GeneratedMethodAccessor12.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at jenkins.ClassLoaderReflectionToolkit.invoke(ClassLoaderReflectionToolkit.java:44)
        at jenkins.ClassLoaderReflectionToolkit._findClass(ClassLoaderReflectionToolkit.java:86)
        at hudson.PluginManager$UberClassLoader.findClass(PluginManager.java:1548)
        at java.lang.<span class="code-object">ClassLoader</span>.loadClass(Unknown Source)
        at java.lang.<span class="code-object">ClassLoader</span>.loadClass(Unknown Source)
        at net.java.sezpoz.IndexItem.element(IndexItem.java:134)
        ... 12 more
Caused by: java.lang.ClassNotFoundException: com.cloudbees.hudson.plugins.folder.AbstractFolderPropertyDescriptor
        at jenkins.util.AntClassLoader.findClassInComponents(AntClassLoader.java:1376)
        at jenkins.util.AntClassLoader.findClass(AntClassLoader.java:1326)
        at jenkins.util.AntClassLoader.loadClass(AntClassLoader.java:1079)
        at java.lang.<span class="code-object">ClassLoader</span>.loadClass(Unknown Source)
        ... 28 more
</pre>
</div></div></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
jenkins 2.0-rc1<br/>
Windows 7<br/>
java version &quot;1.8.0_45&quot;<br/>
```
</details>
<details><summary><i>3 attachments</i></summary>

- [GettingStartedHang.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/32407/GettingStartedHang.png)
  > ![GettingStartedHang.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/32407/GettingStartedHang.png)
- [threadDump.txt](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/32406/threadDump.txt)
- [threadDump2.txt](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/32408/threadDump2.txt)
</details>

**Comment by jenkins-infra-bot** [comment]
- _Remote link associated with this issue:_


  - [PR 2282](https://github.com/jenkinsci/jenkins/pull/2282)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=254169&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254169">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The exception is unrelated. It's a bug in a plugin that results in a minor notification to be presented as a big problem.</p>

<p>Is your internet connection unstable? While this plugin download is going on, are you able to reach e.g. updates.jenkins-ci.org from the same host?</p>
</pre>
</details>

</ul>
</details>

<p>The exception is unrelated. It's a bug in a plugin that results in a minor notification to be presented as a big problem.</p>

<p>Is your internet connection unstable? While this plugin download is going on, are you able to reach e.g. updates.jenkins-ci.org from the same host?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>rg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=254179&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254179">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>&gt; Is your internet connection unstable?</p>

<p>I don't have any other problems with it.</p>

<p>&gt; While this plugin download is going on, are you able to reach e.g. updates.jenkins-ci.org from the same host?</p>

<p>I can reliably ping updates.jenkins-ci.org while it hangs.</p>

<p>I did have both wired and wireless network connections enabled but I've just recreated it with just wired.</p>
</pre>
</details>

</ul>
</details>

<p>> Is your internet connection unstable?</p>

<p>I don't have any other problems with it.</p>

<p>> While this plugin download is going on, are you able to reach e.g. updates.jenkins-ci.org from the same host?</p>

<p>I can reliably ping updates.jenkins-ci.org while it hangs.</p>

<p>I did have both wired and wireless network connections enabled but I've just recreated it with just wired.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=254180&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254180">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Ping isn't enough. May be a problem related to mirrorbrain &#8211; try to download stuff from that site, e.g. plugin files in <a href="http://updates.jenkins-ci.org/latest/" class="external-link" target="_blank" rel="nofollow noopener">http://updates.jenkins-ci.org/latest/</a> &#8211; note which mirror you get HTTP redirected to (if any). </p>
</pre>
</details>

</ul>
</details>

<p>Ping isn't enough. May be a problem related to mirrorbrain &#8211; try to download stuff from that site, e.g. plugin files in <a href="http://updates.jenkins-ci.org/latest/" class="external-link" target="_blank" rel="nofollow noopener">http://updates.jenkins-ci.org/latest/</a> &#8211; note which mirror you get HTTP redirected to (if any). </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=254182&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254182">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The URL <tt>/threadDump</tt> should be manually accessible and tell us what Jenkins is doing while showing this screen. (<a href="https://wiki.jenkins-ci.org/display/JENKINS/Obtaining+a+thread+dump" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Obtaining+a+thread+dump</a>)</p>
</pre>
</details>

</ul>
</details>

<p>The URL <tt>/threadDump</tt> should be manually accessible and tell us what Jenkins is doing while showing this screen. (<a href="https://wiki.jenkins-ci.org/display/JENKINS/Obtaining+a+thread+dump" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Obtaining+a+thread+dump</a>)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>rg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=254186&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254186">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Okay, I have a script which is successfully downloading plugins from <a href="https://updates.jenkins-ci.org/latest/" class="external-link" target="_blank" rel="nofollow noopener">https://updates.jenkins-ci.org/latest/</a>. It has successfully downloaded about 50 and still going. Jenkins seems to have stopped after getting 8. I don't think I've been redirected to a mirror.</p>

<p>Tried a different browser (in case the problem is browser side) and get the same thing.</p>

<p>I attached a thread dump to the initial report, corresponding to the screenshot.</p>

<p>I'll attach threadDump2.txt. This is having downloaded (Ant, OWASP, build timeout, Folders, LDAP, Mailer, Matrix auth and PAM auth).</p>
</pre>
</details>

</ul>
</details>

<p>Okay, I have a script which is successfully downloading plugins from <a href="https://updates.jenkins-ci.org/latest/" class="external-link" target="_blank" rel="nofollow noopener">https://updates.jenkins-ci.org/latest/</a>. It has successfully downloaded about 50 and still going. Jenkins seems to have stopped after getting 8. I don't think I've been redirected to a mirror.</p>

<p>Tried a different browser (in case the problem is browser side) and get the same thing.</p>

<p>I attached a thread dump to the initial report, corresponding to the screenshot.</p>

<p>I'll attach threadDump2.txt. This is having downloaded (Ant, OWASP, build timeout, Folders, LDAP, Mailer, Matrix auth and PAM auth).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>rg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=254187&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254187">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Tail of log</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">←[0mApr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Started initialization
Apr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Listed all plugins
Apr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Prepared all plugins
Apr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Started all plugins
Apr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Augmented all extensions
Apr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Loaded all jobs
Apr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Completed initialization
Apr 12, 2016 4:51:25 PM hudson.ClassicPluginStrategy updateDependency
INFO: Updated dependency of matrix-auth
Apr 12, 2016 4:51:25 PM hudson.PluginManager dynamicLoad
INFO: Plugin cloudbees-folder:5.7 dynamically installed
Apr 12, 2016 4:51:25 PM hudson.model.UpdateCenter$DownloadJob run
INFO: Installation successful: Folders Plugin
Apr 12, 2016 4:51:25 PM hudson.model.UpdateCenter$DownloadJob run
INFO: Starting the installation of Credentials Plugin on behalf of admin
Apr 12, 2016 4:51:26 PM hudson.model.UpdateCenter$UpdateCenterConfiguration down
load
INFO: Downloading Credentials Plugin
Apr 12, 2016 4:51:27 PM hudson.PluginManager dynamicLoad
INFO: Attempting to dynamic load C:\j2\plugins\credentials.jpi
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Started initialization
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Listed all plugins
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Prepared all plugins
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Started all plugins
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Augmented all extensions
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Loaded all jobs
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Completed initialization
Apr 12, 2016 4:51:31 PM hudson.ClassicPluginStrategy updateDependency
INFO: Updated dependency of cloudbees-folder
Apr 12, 2016 4:51:31 PM hudson.PluginManager dynamicLoad
INFO: Plugin credentials:1.27 dynamically installed
Apr 12, 2016 4:51:31 PM hudson.model.UpdateCenter$DownloadJob run
INFO: Installation successful: Credentials Plugin
Apr 12, 2016 4:51:31 PM hudson.model.UpdateCenter$DownloadJob run
INFO: Starting the installation of Structs Plugin on behalf of admin
Apr 12, 2016 4:51:32 PM hudson.model.UpdateCenter$UpdateCenterConfiguration down
load
INFO: Downloading Structs Plugin
Apr 12, 2016 4:56:42 PM hudson.model.AsyncPeriodicWork$1 run
INFO: Started Fingerprint cleanup
Apr 12, 2016 4:56:42 PM hudson.model.AsyncPeriodicWork$1 run
INFO: Finished Fingerprint cleanup. 2 ms
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>Tail of log</p>

<pre>
←[0mApr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Started initialization
Apr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Listed all plugins
Apr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Prepared all plugins
Apr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Started all plugins
Apr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Augmented all extensions
Apr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Loaded all jobs
Apr 12, 2016 4:51:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Completed initialization
Apr 12, 2016 4:51:25 PM hudson.ClassicPluginStrategy updateDependency
INFO: Updated dependency of matrix-auth
Apr 12, 2016 4:51:25 PM hudson.PluginManager dynamicLoad
INFO: Plugin cloudbees-folder:5.7 dynamically installed
Apr 12, 2016 4:51:25 PM hudson.model.UpdateCenter$DownloadJob run
INFO: Installation successful: Folders Plugin
Apr 12, 2016 4:51:25 PM hudson.model.UpdateCenter$DownloadJob run
INFO: Starting the installation of Credentials Plugin on behalf of admin
Apr 12, 2016 4:51:26 PM hudson.model.UpdateCenter$UpdateCenterConfiguration down
load
INFO: Downloading Credentials Plugin
Apr 12, 2016 4:51:27 PM hudson.PluginManager dynamicLoad
INFO: Attempting to dynamic load C:\j2\plugins\credentials.jpi
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Started initialization
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Listed all plugins
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Prepared all plugins
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Started all plugins
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Augmented all extensions
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Loaded all jobs
Apr 12, 2016 4:51:31 PM jenkins.InitReactorRunner$1 onAttained
INFO: Completed initialization
Apr 12, 2016 4:51:31 PM hudson.ClassicPluginStrategy updateDependency
INFO: Updated dependency of cloudbees-folder
Apr 12, 2016 4:51:31 PM hudson.PluginManager dynamicLoad
INFO: Plugin credentials:1.27 dynamically installed
Apr 12, 2016 4:51:31 PM hudson.model.UpdateCenter$DownloadJob run
INFO: Installation successful: Credentials Plugin
Apr 12, 2016 4:51:31 PM hudson.model.UpdateCenter$DownloadJob run
INFO: Starting the installation of Structs Plugin on behalf of admin
Apr 12, 2016 4:51:32 PM hudson.model.UpdateCenter$UpdateCenterConfiguration down
load
INFO: Downloading Structs Plugin
Apr 12, 2016 4:56:42 PM hudson.model.AsyncPeriodicWork$1 run
INFO: Started Fingerprint cleanup
Apr 12, 2016 4:56:42 PM hudson.model.AsyncPeriodicWork$1 run
INFO: Finished Fingerprint cleanup. 2 ms
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>rg:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=254194&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254194">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I wonder if there is a deadlock somewhere in the plugin download code due to higher latency. My ping times to updates.jenkins-ci.org are about 100ms.</p>

<p>Is there an option to download plugins sequentially? Or can I specify a mirror with a lower ping?</p>
</pre>
</details>

</ul>
</details>

<p>I wonder if there is a deadlock somewhere in the plugin download code due to higher latency. My ping times to updates.jenkins-ci.org are about 100ms.</p>

<p>Is there an option to download plugins sequentially? Or can I specify a mirror with a lower ping?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=254292&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254292">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Notable excerpts from the thread dumps:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>"Update center installer thread [#2]: http://updates.jenkins-ci.org/download/plugins/structs/1.1/structs.hpi" Id=80 Group=main RUNNABLE (in native)
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(Unknown Source)
	at java.net.SocketInputStream.read(Unknown Source)
	at java.net.SocketInputStream.read(Unknown Source)
	at java.io.BufferedInputStream.read1(Unknown Source)
	at java.io.BufferedInputStream.read(Unknown Source)
	-  locked java.io.BufferedInputStream@4d998b3c
	at sun.net.www.MeteredStream.read(Unknown Source)
	-  locked sun.net.www.MeteredStream@541aaf13
	at java.io.FilterInputStream.read(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection$HttpInputStream.read(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection$HttpInputStream.read(Unknown Source)
	at org.apache.commons.io.input.ProxyInputStream.read(ProxyInputStream.java:77)
	at hudson.model.UpdateCenter$UpdateCenterConfiguration.download(UpdateCenter.java:1020)
	at hudson.model.UpdateCenter$DownloadJob._run(UpdateCenter.java:1474)
	at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:1655)
	at hudson.model.UpdateCenter$DownloadJob.run(UpdateCenter.java:1452)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:110)
	at java.lang.Thread.run(Unknown Source)</pre>
</div></div>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>"Update center installer thread [#2]: http://updates.jenkins-ci.org/download/plugins/external-monitor-job/1.4/external-monitor-job.hpi" Id=80 Group=main RUNNABLE (in native)
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(Unknown Source)
	at java.net.SocketInputStream.read(Unknown Source)
	at java.net.SocketInputStream.read(Unknown Source)
	at java.io.BufferedInputStream.read1(Unknown Source)
	at java.io.BufferedInputStream.read(Unknown Source)
	-  locked java.io.BufferedInputStream@64d42078
	at sun.net.www.MeteredStream.read(Unknown Source)
	-  locked sun.net.www.MeteredStream@60eba83a
	at java.io.FilterInputStream.read(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection$HttpInputStream.read(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection$HttpInputStream.read(Unknown Source)
	at org.apache.commons.io.input.ProxyInputStream.read(ProxyInputStream.java:77)
	at hudson.model.UpdateCenter$UpdateCenterConfiguration.download(UpdateCenter.java:1020)
	at hudson.model.UpdateCenter$DownloadJob._run(UpdateCenter.java:1474)
	at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:1655)
	at hudson.model.UpdateCenter$DownloadJob.run(UpdateCenter.java:1452)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:110)
	at java.lang.Thread.run(Unknown Source)</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>Notable excerpts from the thread dumps:</p>



```
"Update center installer thread [#2]: http://updates.jenkins-ci.org/download/plugins/structs/1.1/structs.hpi" Id=80 Group=main RUNNABLE (in native)
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(Unknown Source)
	at java.net.SocketInputStream.read(Unknown Source)
	at java.net.SocketInputStream.read(Unknown Source)
	at java.io.BufferedInputStream.read1(Unknown Source)
	at java.io.BufferedInputStream.read(Unknown Source)
	-  locked java.io.BufferedInputStream@​4d998b3c
	at sun.net.www.MeteredStream.read(Unknown Source)
	-  locked sun.net.www.MeteredStream@​541aaf13
	at java.io.FilterInputStream.read(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection$HttpInputStream.read(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection$HttpInputStream.read(Unknown Source)
	at org.apache.commons.io.input.ProxyInputStream.read(ProxyInputStream.java:77)
	at hudson.model.UpdateCenter$UpdateCenterConfiguration.download(UpdateCenter.java:1020)
	at hudson.model.UpdateCenter$DownloadJob._run(UpdateCenter.java:1474)
	at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:1655)
	at hudson.model.UpdateCenter$DownloadJob.run(UpdateCenter.java:1452)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:110)
	at java.lang.Thread.run(Unknown Source)
```



```
"Update center installer thread [#2]: http://updates.jenkins-ci.org/download/plugins/external-monitor-job/1.4/external-monitor-job.hpi" Id=80 Group=main RUNNABLE (in native)
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(Unknown Source)
	at java.net.SocketInputStream.read(Unknown Source)
	at java.net.SocketInputStream.read(Unknown Source)
	at java.io.BufferedInputStream.read1(Unknown Source)
	at java.io.BufferedInputStream.read(Unknown Source)
	-  locked java.io.BufferedInputStream@​64d42078
	at sun.net.www.MeteredStream.read(Unknown Source)
	-  locked sun.net.www.MeteredStream@​60eba83a
	at java.io.FilterInputStream.read(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection$HttpInputStream.read(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection$HttpInputStream.read(Unknown Source)
	at org.apache.commons.io.input.ProxyInputStream.read(ProxyInputStream.java:77)
	at hudson.model.UpdateCenter$UpdateCenterConfiguration.download(UpdateCenter.java:1020)
	at hudson.model.UpdateCenter$DownloadJob._run(UpdateCenter.java:1474)
	at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:1655)
	at hudson.model.UpdateCenter$DownloadJob.run(UpdateCenter.java:1452)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:110)
	at java.lang.Thread.run(Unknown Source)
```

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=254299&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254299">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Should probably timeout a plugin download that just doesn't finish, combined with an option to retry.</p>
</pre>
</details>

</ul>
</details>

<p>Should probably timeout a plugin download that just doesn't finish, combined with an option to retry.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kzantow.png" title="kzantow's avatar" /> kzantow:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=254338&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254338">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This definitely looks like some sort of network issue, we can add some timeouts to the plugin downloads, I think, which would end up causing failures to be reported to the installer (which is the right thing to do). This doesn't look like any deadlock issue to me, as these threads are waiting on responses from the OS sockets:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-none">"Update center installer thread [#2]: http://updates.jenkins-ci.org/download/plugins/structs/1.1/structs.hpi" Id=80 Group=main RUNNABLE (in native)
	at java.net.SocketInputStream.socketRead0(Native Method)
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>This definitely looks like some sort of network issue, we can add some timeouts to the plugin downloads, I think, which would end up causing failures to be reported to the installer (which is the right thing to do). This doesn't look like any deadlock issue to me, as these threads are waiting on responses from the OS sockets:</p>


<pre>
"Update center installer thread [#2]: http://updates.jenkins-ci.org/download/plugins/structs/1.1/structs.hpi" Id=80 Group=main RUNNABLE (in native)
	at java.net.SocketInputStream.socketRead0(Native Method)
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>svanoort:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=254463&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254463">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Encountered in testfest, definitely needs a timeout+retry option, hangs indefinitely for me when there is a network fault. </p>

<p>Can this show the error like we do with normal plugin installation?</p>
</pre>
</details>

</ul>
</details>

<p>Encountered in testfest, definitely needs a timeout+retry option, hangs indefinitely for me when there is a network fault. </p>

<p>Can this show the error like we do with normal plugin installation?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kzantow.png" title="kzantow's avatar" /> kzantow:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=254466&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254466">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=svanoort" class="user-hover" rel="svanoort">svanoort</a> yes, a failed plugin install for any reason should show up as an error. In addition to the 'retry' option from another ticket, I think the experience here should be as good as can be expected when downloading plugins is failing.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=svanoort" class="user-hover" rel="svanoort">svanoort</a> yes, a failed plugin install for any reason should show up as an error. In addition to the 'retry' option from another ticket, I think the experience here should be as good as can be expected when downloading plugins is failing.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>svanoort:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=254489&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254489">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yeah, if there's any way we can simply use the logic from normal plugin installation, I think that's a reasonable step, if it saves effort. </p>
</pre>
</details>

</ul>
</details>

<p>Yeah, if there's any way we can simply use the logic from normal plugin installation, I think that's a reasonable step, if it saves effort. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=255587&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-255587">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: kzantow
Path:
 core/src/main/java/hudson/PluginManager.java
 core/src/main/java/hudson/model/UpdateCenter.java
 core/src/main/resources/jenkins/install/pluginSetupWizard.properties
 war/src/main/js/api/pluginManager.js
 war/src/main/js/pluginSetupWizardGui.js
 war/src/main/js/templates/successPanel.hbs
<a href="http://jenkins-ci.org/commit/jenkins/6974d2b7593e3eef3a7fd28117b58382c3368189" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/6974d2b7593e3eef3a7fd28117b58382c3368189</a>
Log:
  <a href="https://issues.jenkins.io/browse/JENKINS-33244" title="Initial plugin installation should have retry for failed plugins" class="issue-link" data-issue-key="JENKINS-33244"><del>JENKINS-33244</del></a> / <a href="https://issues.jenkins.io/browse/JENKINS-34174" title="Jenkins 2.0-rc1 - Getting started hangs installing plugins" class="issue-link" data-issue-key="JENKINS-34174"><del>JENKINS-34174</del></a> - retry for failed plugins &amp; specify
download timeout</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: kzantow<br/>
Path:<br/>
 core/src/main/java/hudson/PluginManager.java<br/>
 core/src/main/java/hudson/model/UpdateCenter.java<br/>
 core/src/main/resources/jenkins/install/pluginSetupWizard.properties<br/>
 war/src/main/js/api/pluginManager.js<br/>
 war/src/main/js/pluginSetupWizardGui.js<br/>
 war/src/main/js/templates/successPanel.hbs<br/>
<a href="http://jenkins-ci.org/commit/jenkins/6974d2b7593e3eef3a7fd28117b58382c3368189" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/6974d2b7593e3eef3a7fd28117b58382c3368189</a><br/>
Log:<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-33244" title="Initial plugin installation should have retry for failed plugins" class="issue-link" data-issue-key="JENKINS-33244"><del>JENKINS-33244</del></a> / <a href="https://issues.jenkins.io/browse/JENKINS-34174" title="Jenkins 2.0-rc1 - Getting started hangs installing plugins" class="issue-link" data-issue-key="JENKINS-34174"><del>JENKINS-34174</del></a> - retry for failed plugins & specify<br/>
download timeout</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=255589&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-255589">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: kzantow
Path:
 core/src/main/java/hudson/PluginManager.java
 core/src/main/resources/jenkins/install/pluginSetupWizard.properties
 war/src/main/js/pluginSetupWizardGui.js
 war/src/main/js/templates/pluginSelectionPanel.hbs
 war/src/main/js/templates/welcomePanel.hbs
 war/src/main/less/pluginSetupWizard.less
<a href="http://jenkins-ci.org/commit/jenkins/e612f4163a0e1acedfed9baed6e79cf2bad6fabe" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/e612f4163a0e1acedfed9baed6e79cf2bad6fabe</a>
Log:
  Merge remote-tracking branch 'primary/master' into
<a href="https://issues.jenkins.io/browse/JENKINS-34174" title="Jenkins 2.0-rc1 - Getting started hangs installing plugins" class="issue-link" data-issue-key="JENKINS-34174"><del>JENKINS-34174</del></a>-plugin-download-timeout</p>

<ol>
	<li>Conflicts:</li>
	<li>core/src/main/resources/jenkins/install/pluginSetupWizard.properties</li>
</ol>



</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: kzantow<br/>
Path:<br/>
 core/src/main/java/hudson/PluginManager.java<br/>
 core/src/main/resources/jenkins/install/pluginSetupWizard.properties<br/>
 war/src/main/js/pluginSetupWizardGui.js<br/>
 war/src/main/js/templates/pluginSelectionPanel.hbs<br/>
 war/src/main/js/templates/welcomePanel.hbs<br/>
 war/src/main/less/pluginSetupWizard.less<br/>
<a href="http://jenkins-ci.org/commit/jenkins/e612f4163a0e1acedfed9baed6e79cf2bad6fabe" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/e612f4163a0e1acedfed9baed6e79cf2bad6fabe</a><br/>
Log:<br/>
  Merge remote-tracking branch 'primary/master' into<br/>
<a href="https://issues.jenkins.io/browse/JENKINS-34174" title="Jenkins 2.0-rc1 - Getting started hangs installing plugins" class="issue-link" data-issue-key="JENKINS-34174"><del>JENKINS-34174</del></a>-plugin-download-timeout</p>

<ol>
	<li>Conflicts:</li>
	<li>core/src/main/resources/jenkins/install/pluginSetupWizard.properties</li>
</ol>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-34174?focusedId=255590&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-255590">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Daniel Beck
Path:
 core/src/main/java/hudson/PluginManager.java
 core/src/main/java/hudson/model/UpdateCenter.java
 core/src/main/resources/jenkins/install/pluginSetupWizard.properties
 war/src/main/js/api/pluginManager.js
 war/src/main/js/pluginSetupWizardGui.js
 war/src/main/js/templates/successPanel.hbs
<a href="http://jenkins-ci.org/commit/jenkins/8a19e8ae3d82602bf36ac5624056cae2e6950e25" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/8a19e8ae3d82602bf36ac5624056cae2e6950e25</a>
Log:
  Merge pull request #2282 from kzantow/<a href="https://issues.jenkins.io/browse/JENKINS-34174" title="Jenkins 2.0-rc1 - Getting started hangs installing plugins" class="issue-link" data-issue-key="JENKINS-34174"><del>JENKINS-34174</del></a>-plugin-download-timeout</p>

<p><a href="https://issues.jenkins.io/browse/JENKINS-33244" title="Initial plugin installation should have retry for failed plugins" class="issue-link" data-issue-key="JENKINS-33244"><del>JENKINS-33244</del></a> better handling of failed plugins during setup wizard</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/a4b070bd7357...8a19e8ae3d82" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/a4b070bd7357...8a19e8ae3d82</a></p>

</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Daniel Beck<br/>
Path:<br/>
 core/src/main/java/hudson/PluginManager.java<br/>
 core/src/main/java/hudson/model/UpdateCenter.java<br/>
 core/src/main/resources/jenkins/install/pluginSetupWizard.properties<br/>
 war/src/main/js/api/pluginManager.js<br/>
 war/src/main/js/pluginSetupWizardGui.js<br/>
 war/src/main/js/templates/successPanel.hbs<br/>
<a href="http://jenkins-ci.org/commit/jenkins/8a19e8ae3d82602bf36ac5624056cae2e6950e25" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/8a19e8ae3d82602bf36ac5624056cae2e6950e25</a><br/>
Log:<br/>
  Merge pull request #2282 from kzantow/<a href="https://issues.jenkins.io/browse/JENKINS-34174" title="Jenkins 2.0-rc1 - Getting started hangs installing plugins" class="issue-link" data-issue-key="JENKINS-34174"><del>JENKINS-34174</del></a>-plugin-download-timeout</p>

<p><a href="https://issues.jenkins.io/browse/JENKINS-33244" title="Initial plugin installation should have retry for failed plugins" class="issue-link" data-issue-key="JENKINS-33244"><del>JENKINS-33244</del></a> better handling of failed plugins during setup wizard</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/a4b070bd7357...8a19e8ae3d82" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/a4b070bd7357...8a19e8ae3d82</a></p>
